### PR TITLE
Check zone id before jdbc access

### PR DESCRIPTION
**Pull request by strehle**
After discussions in https://github.com/cloudfoundry/uaa/pull/2606

Add null checks to improve robustness in this dao using validateZoneId helper.

Cleanup sonar issues, found by IDE plugin

https://sonarcloud.io/summary/new_code?id=cloudfoundry-identity-parent&pullRequest=2616

### CODE REVIEW

File: server/src/main/java/org/cloudfoundry/identity/uaa/scim/jdbc/JdbcScimGroupProvisioning.java [hunk 1]
```diff
@@ -185,14 +186,11 @@ public ScimGroup getByName(String displayName, String zoneId) {
...
     public void onApplicationEvent(AbstractUaaEvent event) {
-        if (event instanceof IdentityZoneModifiedEvent) {
-            IdentityZoneModifiedEvent zevent = (IdentityZoneModifiedEvent) event;
-            if (zevent.getEventType() == AuditEventType.IdentityZoneCreatedEvent) {
-                final String zoneId = ((IdentityZone) event.getSource()).getId();
-                getSystemScopes().forEach(
-                        scope -> createAndIgnoreDuplicate(scope, zoneId)
-                );
-            }
+        if (event instanceof IdentityZoneModifiedEvent zevent && zevent.getEventType() == AuditEventType.IdentityZoneCreatedEvent) {
```
Comment (strehle) [inline server/src/main/java/org/cloudfoundry/identity/uaa/scim/jdbc/JdbcScimGroupProvisioning.java:189]: java 16 feature

**Review by klaus-sap — APPROVED** [review]
looks good to me

**Comment by strehle** [comment]
@swalchemist similar to zone check here a small PR which does a group id check before JDBC call.

**Review by tack-sap — APPROVED** [review]
LGTM

**Pull request merged**